feature/itinerary/domain/src/main/java/com/serranoie/app/feature/itinerary/domain/usecase/ItineraryUseCase.kt

```
@@ -0,0 +1,80 @@
+        request: CreateItineraryItem
+    ): Result<ItineraryItem> {
+        return repository.createActivity(groupCode, request)
+    }
+}
+
+class DeleteActivityByIdUseCase(private val repository: ItineraryRepository) {
+    suspend operator fun invoke(itineraryId: String): Result<Unit> {
+        return try {
+            repository.deleteActivityById(itineraryId)
+            Result.Success(Unit)
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)